site/source/docs/porting/simd.rst

```diff
@@ -1015,3 +1015,301 @@ The following table highlights the availability and expected performance of diff
+
+In terms of performance, it is very important to note that only
+instructions which operate on 128-bit wide vectors are supported
+cleanly. This means that nearly any instruction which is not of a "q"
+variant (i.e. "vaddq" as opposed to "vadd") will be scalarized.
+
+These are pulled from `SIMDe repository on Github
+<https://github.com/simd-everywhere/simde>`_. To update emscripten
+with the latest SIMDe version, run `tools/simde_update.py`.
+
+The following table highlights the availability of various 128-bit
+wide intrinsics.
+
+Similarly to above, the following legend is used:
+ - ✅ Wasm SIMD has a native opcode that matches the NEON instruction, should yield native performance
+ - 💡 while the Wasm SIMD spec does not provide a proper performance guarantee, given a suitably smart enough compiler and a runtime VM path, this intrinsic should be able to generate the identical native NEON instruction.
+ - ⚠️ the underlying NEON instruction is not available, but it is emulated via at most few other Wasm SIMD instructions, causing a small penalty.
+ - ❌ the underlying NEON instruction is not exposed by the Wasm SIMD specification, so it must be emulated via a slow path, e.g. a sequence of several slower SIMD instructions, or a scalar implementation.
+ - ⚫ the given NEON intrinsic is not available. Referencing the intrinsic will cause a compiler error.
+
+For detailed information on each intrinsic function, refer to `NEON Intrinsics Reference
+<https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics>`_.
+
+.. list-table:: NEON Intrinsics
+   :widths: 20 30
+   :header-rows: 1
+
+   * - Intrinsic name
+     - Wasm SIMD Support
+   * - vaba
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabal
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabd
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabdl
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabs
+     - native
+   * - vadd
+     - native
+   * - vaddl
+     - ⚫ Not implemented, will trigger compiler error
+   * - vaddlv
+     - ⚫ Not implemented, will trigger compiler error
+   * - vaddv
+     - ⚫ Not implemented, will trigger compiler error
+   * - vaddw 
+     - ❌ Will be emulated with slow instructions, or scalarized
+   * - vand 
+     - native
+   * - vbic
+     - ⚫ Not implemented, will trigger compiler error
+   * - vbsl
+     - native
+   * - vcagt
+     - ⚠ Does not have direct implementation, but is emulated using fast NEON instructions
+   * - vceq
+     - 💡 Depends on a smart enough compiler, but should be near native
+   * - vceqz
+     - ⚠ Does not have direct implementation, but is emulated using fast NEON instructions
+   * - vcge
+     - native
+   * - vcgez
+     - ⚠ Does not have direct implementation, but is emulated using fast NEON instructions
+   * - vcgt
+     - native
+   * - vcgtz
+     - ⚠ Does not have direct implementation, but is emulated using fast NEON instructions
```

Comment:
Taking e.g. `vcgt` and `vcgtz` for a closer look, this seems inaccurate also. More precisely, these look like:

```
 * - vcgt_f32, vcgt_f64, vcgt_s8, vcgt_s16, vcgt_s32, vcgt_s64, vcgtz_f32, vcgtz_f64, vcgtz_s8, vcgtz_s16, vcgtz_s32, vcgtz_s64
   - ❌ Scalarized via operations on 64-bit wide structs.
 * - vcgtq_f32, vcgtq_f64, vcgtq_s8, vcgtq_s16, vcgtq_s32, vcgtq_u8, vcgtq_u16, vcgtq_u32
   - ✅ wasm_f32x4_gt, wasm_f64x2_gt, wasm_i8x16_gt, wasm_i16x8_gt, wasm_i32x4_gt, wasm_u8x16_gt, wasm_u16x8_gt, wasm_u32x4_gt
 * - vcgtq_s64, vcgtq_u64
   - ❌ Scalarized.
 * - vcgtz_f32, vcgtz_f64, vcgtz_s8, vcgtz_s16, vcgtz_s32, vcgtz_s64
   - ❌ Scalarized.
 * - vcgtzq_f32, vcgtzq_f64, vcgtzq_s8, vcgtzq_s16, vcgtzq_s32
   - ⚠️ Emits a combination of vcgtq_<t> + vdupq_n_<t>
 * - vcgtzq_s64
   - ❌ Scalarized.
```

The use of `simde_vdupq_n_f32(SIMDE_FLOAT32_C(0.0))` in `vcgtzq_f32` is a bit iffy, would be nice to prefer `wasm_f32x4_const(0, 0, 0, 0)` everywhere when literals are used, rather than possibly runtime constructs. That way the best code will be emitted independent of the optimization&tuning flags that are used.